annotated.go

```diff
@@ -538,9 +1000,24 @@ func OnStop(onStop interface{}) Annotation {
 
 type asAnnotation struct {
 	targets []interface{}
+	types   []reflect.Type
+}
+
+func isOut(t reflect.Type) bool {
+	return (t.Kind() == reflect.Struct &&
+		dig.IsOut(reflect.New(t).Elem().Interface()))
 }
 
-var _ Annotation = asAnnotation{}
+func isIn(t reflect.Type) bool {
+	return (t.Kind() == reflect.Struct &&
+		dig.IsIn(reflect.New(t).Elem().Interface()))
+}
+
+func checkIfResultTagged(results []reflect.Type) (reflect.Type, bool) {
```

Comment:
technically this is just checking whether the result is an fx.Out struct, so if the function is producing an fx.Out struct but is not tagged with ResultTags, it'll still return true from this func, right?

Reply:
That's correct. However, since the original function being annotated  cannot have any Out struct in its return (checked by `typeCheckOrigFn()`), it's safe to assume that an `fx.ResultTags` was applied if the first return type is an `fx.Out`. Obviously this might change if we start allowing the original function being annotated to have an `fx.Out` struct in its return values, but that's for later.

Reply:
That being said, just realized this is used only in one place. I'm just going to remove this function and inline the logic where it's used.